src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -270,8 +328,8 @@ SDCPUThread::bufferInstructionData(Addr vaddr, uint8_t* data)
     DPRINTF(SDCPUThreadEvent, "Updating fetch buffer with data from %#x\n",
                               vaddr);
 
+    assert(vaddr);
     fetchBufBase = vaddr;
-    if (!vaddr) return;
```

Comment:
It was done this way before, to provide a function call which would clear the buffer.